src/model/file_test.go

```diff
@@ -0,0 +1,106 @@
+package model
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+func TestReadYAML(t *testing.T) {
+	type testCase struct {
+		name string
+		yaml string
+		cfg  *Config
+		err  error
+	}
+
+	tmpdir := t.TempDir()
+	testCases := []testCase{
+		{
+			name: "FileDoesNotExist",
+			yaml: "",
+			cfg:  nil,
+			err:  errors.New("failed to read file"),
+		},
+		{
+			name: "InvalidYAML",
+			yaml: `: this is not valid yaml`,
+			cfg:  nil,
+			err:  errors.New("failed to unmarshal data"),
+		},
+		{
+			name: "NilConfig",
+			yaml: `{"kernel_cmdline":{{}`,
+			cfg:  nil,
+			err:  errors.New("failed to unmarshal data"),
+		},
```

Comment:
```suggestion
		{
			name: "InvalidYAML",
			yaml: `: this is not valid yaml`,
			cfg:  nil,
			err:  errors.New("failed to unmarshal data"),
		},
```
The second one is redundant if the goal is to test if yaml parsing errors are captured.